lsp-completion.el

```diff
@@ -510,7 +520,18 @@ Others: CANDIDATES"
          (text-edit?
           (apply #'delete-region markers)
           (insert prefix)
-          (lsp--apply-text-edit text-edit?))
+          (pcase text-edit?
+            ((TextEdit) (lsp--apply-text-edit text-edit?))
+            ((InsertReplaceEdit :insert :replace :new-text)
+             (lsp--apply-text-edit
+              (lsp-make-text-edit
+               :new-text new-text
+               :range (if (or (and current-prefix-arg
```

Comment:
Shouldn't this be
``` elisp
(if (or (and current-prefix-arg
             (eq lsp-completion-default-behaviour :replace))
        (and (not current-prefix-arg)
             (eq lsp-completion-default-behaviour :insert)))
    insert
  replace)
```

So without current prefix, it will execute `lsp-completion-default-behavior` but will do otherwise with current prefix.
Currently it will execute `lsp-completion-default-behavior` with `C-u` but do otherwise without current prefix, which is strange to the name and doc of `lsp-completion-default-behavior`

Reply:
Given that there's only `:replace` or `:insert` of `lsp-completion-default-behavior` and also binary state of `current-prefix-arg`. You can probably just use `xor` here too (nit)
``` elisp
if (xor current-prefix-arg (eq lsp-completion-default-behavior :insert)) insert replace)
```

Reply:
Yes, will do. I will change the default as well. Seems like replace is closer to what we have now.

Reply:
xor is added in emacs 27.1...